v2/googlecloud-to-elasticsearch/docs/BigQueryToElasticsearch/README.md

```diff
@@ -1,14 +1,35 @@
 # BigQuery to Elasticsearch Dataflow Template
 
 The [BigQueryToElasticsearch](../../src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/BigQueryToElasticsearch.java) pipeline ingests
-data from a BigQuery table into Elasticsearch. The template can either read the entire table or read using a supplied query.
+data from a BigQuery table into Elasticsearch as a batch job. BigQuery users can ingest data into Elasticsearch and then query and analyze the search results using Elasticsearch APIs or Kibana. The template can either read the entire table or read using a supplied query and ingest the results to Elasticsearch index.
 
-Pipeline flow is illustrated below:
-
-![alt text](../img/bigquery-to-elasticsearch-dataflow.png "BigQuery to Elasticsearch pipeline flow")
+## Use Cases
+It’s common to utilize the output of BigQuery SQL jobs to create further views and tables in BigQuery, or create dashboards to share with other stakeholders and teams in your organization — of which can be achieved with Kibana, Elastic’s native data visualization tool! Another key use case for BigQuery and the Elastic Stack is a full-text search. You can now search data in BigQuery with a few clicks by using the Dataflow template.
 
 ## Getting Started
 
+The simplest way to set up Elasticsearch is to create a managed deployment with Elasticsearch Service on [Elastic Cloud](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started.html#run-elasticsearch). You can deploy Elasticsearch on many [Google Cloud regions](https://www.elastic.co/guide/en/cloud/current/ec-regions-templates-instances.html) globally. If you prefer to manage your own environment, you can install and run [Elasticsearch using Docker](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started.html#run-elasticsearch).
+
+### Parameters
+
+The template requires the following parameters:
+* inputTableSpec: Table in BigQuery to read from in form of: my-project:my-dataset.my-table. Either this or query must be provided.
```

Comment:
inputTableSpec is also optional parameter since user can supply either query or inputTableSpec